### ISSUE TITLE
Feature request: Add support for deployment_type in apigee module

### ISSUE THREAD

**Issue by blackillzone** [GoogleCloudPlatform/cloud-foundation-fabric#1114]
Hello,

Since the provider support the [deployment_type](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/apigee_environment#deployment_type) argument for Apigee environment deployments (https://github.com/hashicorp/terraform-provider-google/pull/11405), it would be a nice upcoming add for the apigee module.

In the same time, we could also add the [api_proxy_type](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/apigee_environment#api_proxy_type) parameter.

I guess it would be something like this for the module variables:

```hcl
environments = {
    apis-test = {
      display_name = "APIs test"
      description  = "APIs Test"
      deployment_type = "PROXY"
      api_proxy_type = "PROGRAMMABLE"
      envgroups    = ["test"]
    }
}
```

And I suppose it should be optional, with default values at null, and if null, we don't set the options in the `google_apigee_environment` resource.

This way, the current module should be without breaking changes.

**Comment by blackillzone** [comment]
Added https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/pull/1115, and waiting for the CLA to be validated.

**Comment by juliocc** [comment]
Fixed with #1115
